src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final DateTimeFormatter FORMATTER =
+            DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_FROM_DATA = 0;
+    private static final int NAME_FROM_DATA = 1;
+    private static final int HOURS_FROM_DATA = 2;
+    private static final int SALARY_FROM_DATA = 3;
+
+    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
+            throws DateTimeException {
+        LocalDate startDate;
+        LocalDate endDate;
+        LocalDate dateInArray;
+        StringBuilder salaryInfo = new StringBuilder();
+        try {
+            startDate = LocalDate.parse(dateFrom, FORMATTER);
+            endDate = LocalDate.parse(dateTo, FORMATTER);
+        } catch (DateTimeException e) {
+            throw new DateTimeException("Cannot parse date. Format of date should be"
+                    + "dd.mm.yyyy");
+        }
```

Comment:
let's remove try catch 
```suggestion
            startDate = LocalDate.parse(dateFrom, FORMATTER);
            endDate = LocalDate.parse(dateTo, FORMATTER);       
```